packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,225 @@
+# Bottom Sheet
+
+Bottom sheet is a container that presents additional information to a user in an overlay.
+
+This document outlines the API of `BottomSheet` component.
+
+<img src="./bottom-sheet-thumbnail.png" width="380" alt="" />
+
+## Design
+
+- [Figma - BottomSheet](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=26477%3A578228&t=1RMqjs99P33Udj0C-0)
+
+## API
+
+Sample usage:
+
+```jsx
+import { BottomSheet } from '@razorpay/blade';
+
+<BottomSheet open={boolean} snapPoints={[]}>
+  <BottomSheetHeader>
+    <BottomSheetHeaderLeading title="Payments Links" prefix={BladeIcon} />
+    <BottomSheetHeaderTrailing visual={Link | Action | Icon} />
+  </BottomSheetHeader>
+  <BottomSheetBody>Body Content</BottomSheetBody>
+  <BottomSheet>
+    <BottomSheetLeading title="Footer Title" prefix={icon} />
+    <BottomSheetTrailing
+      actions={{
+        primaryAction: { text: 'Know more', onClick: () => {} },
+        secondaryAction: { text: 'Read Docs', onClick: () => {} },
+      }}
+    />
+  </BottomSheet>
+</BottomSheet>;
+```
+
+**Design Constraints**
+
+- BottomSheetHeader:
+
+  - Leading:
+    - Title
+    - Prefix
+  - Trailing:
+    - empty
+    - icon
+    - link
+    - action
+
+- BottomSheetFooter:
+
+  - Leading
+    - title
+    - prefix
+  - Trailing
+    - Primary action button
+    - Secondary action button
+
+- The BottomSheetFooter will behave similar to Card's footer.
+- The Header, Footer can be individually omitted
+
+### `BottomSheet`
+
+We'll expose an `BottomSheeet` component with the following API:
+
+| Prop            | Type        | Default                 | Description                                                                                     | Required |
+| --------------- | ----------- | ----------------------- | ----------------------------------------------------------------------------------------------- | -------- |
+| open            | `boolean`   | `false`                 | toggles bottom sheet state content                                                              |          |
+| snapPoints      | `string[]`  | `['25%', '50%', '85%']` | snappoints in which the bottom sheeet will rest on                                              |          |
+| onDismiss       | `Callback`  | `undefined`             | called when the bottom sheet is closed, either by user state, hitting `esc` or tapping backdrop |          |
+| initialFocusRef | `React.Ref` | `undefined`             | ref element you want to get keyboard focus when opening the sheet                               |          |
```

Comment:
What would this be used for?

Reply:
BottomSheet's a11y is like a modal, we need to provide users with an option to focus on an element when they open the modal, for example they might choose to focus on a primary button first. 

Internally we can also automatically focus on the first "tabbable" dom element inside the BottomSheetBody (need to check the feasibility with ActionList)

Reply:
Do we need to expose this functionality 🤔 

What could be a possible usecase?

Reply:
> What could be a possible usecase?

check this [ariakit example](https://ariakit.org/examples/dialog-react-router) the textarea is focused initially when the modal opens.

This pattern is common in most modals too, for example go to the x dashboard and click "Add Contact" 
<img width="170" alt="image" src="https://user-images.githubusercontent.com/35374649/222060757-9e9758bf-66ae-4867-9936-dddd9fac751d.png">

You'll see that when the modal opens the "Contact Name" field is focused. 

> Do we need to expose this functionality 🤔

Probably we can avoid exposing this and just take the first focusable element as "initial focus" but there are also certain UX rules which users won't be able to do which is outlined in the [wai-aria notes](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/): 
<img width="934" alt="image" src="https://user-images.githubusercontent.com/35374649/222061656-612075ac-e729-4f82-adff-acd46a2de4d1.png">